arch/arm/src/stm32h7/stm32_spi.c

```diff
@@ -1768,23 +1840,48 @@ static void spi_exchange(FAR struct spi_dev_s *dev, FAR const void *txbuffer,
     }
 
 #ifdef CONFIG_STM32H7_DMACAPABLE
-  stm32_dmacfg_t dmacfg1;
-  stm32_dmacfg_t dmacfg2;
+  stm32_dmacfg_t dmatxcfg;
+  stm32_dmacfg_t dmarxcfg;
+
+  /* Setup DMAs */
+
+  /* If this bus uses a in driver buffers we will incur 2 copies,
+   * The copy cost is << less the non DMA transfer time and having
+   * the buffer in the driver ensures DMA can be used. This is bacause
+   * the API does not support passing the buffer extent so the only
+   * extent is buffer + the transfer size. These can sizes be less than
+   * the cache line size, and not aligned and tyicaly greater then 4
+   * bytes, which is about the break even point for the DMA IO overhead.
+   */
+
+  if (txbuffer && priv->txbuf)
+    {
+      if (nbytes > priv->buflen)
```

Comment:
Is it safe to truncatethe buffer?

Reply:
The API does allow for returning an error.  We could change that but in real use the assumption is the allocation is the worst case for the system on that bus.

Reply:
The error is further down. I will force push....

Reply:
Instead of returning error, it's better to:
1.Send the buffer segment by segment or
2.Switch to PIO mode

Reply:
@xiaoxiang781216 - thanks - force pushed using truncated value in DMA setup

Reply:
@xiaoxiang781216 - I would be better. But this is for performance on small packets. The assumption is the worst case size is set for the bus. PIO would defeat the purpose. Looping would also.